solutions/cf-1557B.mdx

```diff
@@ -0,0 +1,343 @@
+
+<center>
+<table>
+	<tbody>
+		<tr>
+			<th>Index:</th>
+			<th className="bg-yellow-100 dark:bg-yellow-800">0</th>
+			<th>1</th>
+			<th>2</th>
+			<th>3</th>
+			<th>4</th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{arr}[i]$</td>
+			<td>6</td>
+			<td>3</td>
+			<td>4</td>
+			<td>2</td>
+			<td>1</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+<center>
+<table>
+	<tbody>
+		<tr>
+			<th>Index:</th>
+			<th className="bg-yellow-100 dark:bg-yellow-800">0</th>
+			<th className="bg-green-100 dark:bg-green-800">1</th>
+			<th>2</th>
+			<th>3</th>
+			<th>4</th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{arr}[i]$</td>
+			<td>6</td>
+			<td>3</td>
+			<td>4</td>
+			<td>2</td>
+			<td>1</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+<center>
+<table>
+	<tbody>
+		<tr>
+			<th>Index:</th>
+			<th className="bg-yellow-100 dark:bg-yellow-800">0</th>
+			<th className="bg-green-100 dark:bg-green-800">1</th>
+			<th className="bg-green-100 dark:bg-green-800">2</th>
+			<th>3</th>
+			<th>4</th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{arr}[i]$</td>
+			<td>6</td>
+			<td>3</td>
+			<td>4</td>
+			<td>2</td>
+			<td>1</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+<center>
+<table>
+	<tbody>
+		<tr>
+			<th>Index:</th>
+			<th className="bg-yellow-100 dark:bg-yellow-800">0</th>
+			<th className="bg-green-100 dark:bg-green-800">1</th>
+			<th className="bg-green-100 dark:bg-green-800">2</th>
+			<th className="bg-red-100 dark:bg-red-800">3</th>
+			<th>4</th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{arr}[i]$</td>
+			<td>6</td>
+			<td>3</td>
+			<td>4</td>
+			<td>2</td>
+			<td>1</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+<center>
+<table>
+	<tbody>
+		<tr>
+			<th>Index:</th>
+			<th className="bg-yellow-100 dark:bg-yellow-800">0</th>
+			<th className="bg-green-100 dark:bg-green-800">1</th>
+			<th className="bg-green-100 dark:bg-green-800">2</th>
+			<th className="bg-red-100 dark:bg-red-800">3</th>
+			<th className="bg-blue-100 dark:bg-blue-800">4</th>
+		</tr>
+	</tbody>
+	<tbody>
+		<tr>
+			<td>$\texttt{arr}[i]$</td>
+			<td>6</td>
+			<td>3</td>
+			<td>4</td>
+			<td>2</td>
+			<td>1</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+Thus, we need 4 subarrays at minimum.
+
+**Time Complexity:** $\mathcal{O}(N\log N)$
+
+<Spoiler title="Time Complexity Proof">
+Sorting Requires $\mathcal{O}(N\log N)$, and repeated map insertions also require $\mathcal{O}(N\log N)$. Hence the overall time complexity is $\mathcal{O}(N\log N)$.
+</Spoiler>
+	
+<LanguageSection>
+<CPPSection>
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+typedef long long ll;
+#define pb push_back
+#define FF first
+#define SS second
+ 
+void solve() {
+	int n, k; cin >> n >> k;
+	int arr[n];
+	int sarr[n];
+	for (int i = 0; i < n; i++) {
+		cin >> arr[i];
+		sarr[i] = arr[i];
+	}
+	map<int, int> next; // Map from one value to the value after it in the sorted array
```

Comment:
same policy w/ comments here - 2 spaces after the actual line of code
also line length? did you check?